### PR TITLE
Add support for mysqld_exporter version 0.11.0

### PR DESCRIPTION
**Pull request by TheMeier**
#### Pull Request (PR) description
Update mysqld_exporter version to 0.11.0
mysqld_exporter flags now use the Kingpin library, and require double-dashes. https://github.com/prometheus/mysqld_exporter/pull/222

Breaking change: if you do not set a version for mysqld_exporter *and* set `extra_options` make sure
 you update `extra_options` to be compatible with version 0.11.0 and greater.

### CODE REVIEW

File: manifests/mysqld_exporter.pp [hunk 1]
```diff
@@ -137,7 +137,7 @@
     notify  => $notify_service,
   }
 
-  $options = "-config.my-cnf=${cnf_config_path} ${extra_options}"
+  $options = "--config.my-cnf==${cnf_config_path} ${extra_options}"
```
Comment (bastelfreak) [inline manifests/mysqld_exporter.pp]: this change would make it backwards incompatible. Can you implement a selector statement that uses the old or the new syntax, depending on the version?

**Comment by bastelfreak** [comment]
Hi @TheMeier, thanks for the PR. Can you please take a look at the inline comment I made? Also please rebase and take a look at the used email address. It isn't associated with your github account.

**Comment by TheMeier** [comment]
@bastelfreak i am not so sure this can be done in a non backward-breaking way. The point is users could be using `$extra_options` and those would have to be changed by the user. 
I put the code in anyway...

**Comment by TheMeier** [comment]
Also rebased..

**Comment by TheMeier** [comment]
ping

**Comment by alexjfisher** [comment]
Whatever we do here, we should probably also do on https://github.com/voxpupuli/puppet-prometheus/pull/179

File: data/defaults.yaml [hunk 2]
```diff
@@ -168,7 +168,7 @@ prometheus::mysqld_exporter::group: 'mysqld-exporter'
 prometheus::mysqld_exporter::package_ensure: 'latest'
 prometheus::mysqld_exporter::package_name: 'mysqld_exporter'
 prometheus::mysqld_exporter::user: 'mysqld-exporter'
-prometheus::mysqld_exporter::version: '0.9.0'
+prometheus::mysqld_exporter::version: '0.11.0'
```
[the remark this answers is not in the document]
Comment (TheMeier) [reply data/defaults.yaml]: This sounds like a good compromise to me. It users to upgrade withought changing any of the current behaviour.

**Comment by bastelfreak** [comment]
@TheMeier thanks for the updates. Can you please add an acceptance test that installs version `0.9.0` and afterwards bumps it to `0.11.0`? Let us know if you need any help. You can find us in #voxpupuli on the freenode IRC network and on https://puppetcommunity.slack.com.

**Comment by TheMeier** [comment]
@bastelfreak done

**Comment by alexjfisher** [comment]
@TheMeier Thanks!

**Pull request merged**